lib/game_play.rb

```diff
@@ -59,25 +60,37 @@ def create_players(selection)
     end
   end
 
+  def get_user_selection_between(min, max)
+    value = get_user_selection
+    if !value.between?(min, max)
+      @ui.between(min, max)
+      value = get_user_selection_between(min, max)
```

Comment:
The recursion here is cool.  I tried something similar in Java, but found if an invalid choice was made and the method had to recurse, the user would have to enter a valid option twice: once to exit current loop and then again to exit the original.  I guess recursion in ruby must work differently!